### PR TITLE
Add definition of ready and done

### PR DESCRIPTION
**Pull request by jessiehuff**
Closes https://issues.redhat.com/browse/RHOAIENG-10271 

## Description
We need our [Definition of Done](https://docs.google.com/document/d/1A3Y4P04MVlovOK1XvUZL8SpQ0wY94LwQsxe8DwEGoGE/edit) and [Definition of Ready](https://docs.google.com/document/d/1JP8FlysPrsEH5mfx4Jq6t-jr1XBIORS5PoRpeSMkNt8/edit) docs moved into markdown files in the docs folder and our contribution markdown file updated to point at those

## How Has This Been Tested?
N/A (Non-code change)

## Test Impact
N/A

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: docs/definition-of-done.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+- UI visual changes match designs proposed by UX.
+
+## Testing and verification
+- For feature work - a demo has been created and recorded (and linked to in the relevant Jira issue).  For bugs - screenshots should be added to the pull request and linked to from Jira for reference.
+- All changes (fixes, new features...) have been tested at least manually (ideally automated) using a built image.  This can be a nightly ODH image or an RC build.
+- Information about how the changes were tested should be added to the relevant Jira issue (epic, bug or story with no epic parent).
+- All code has been merged to the downstream branch and automated quality gates have successfully passed.
+
+## Jira status
+- Screenshots of the updated UI should be captured in the jira history/comments (or link to GitHub Pull Request with these screenshots) to illustrate work that was completed.
```
Comment (Griffin-Sullivan) [inline docs/definition-of-done.md:18]:
```suggestion
- Screenshots of the updated UI should be captured in the Jira history/comments (or link to GitHub Pull Request with these screenshots) to illustrate work that was completed.
```
Jira was capitalized everywhere else

File: docs/definition-of-ready.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+9. Changes to requirements during the design, development, and testing phases must be promptly communicated and justified to impacted stakeholders who must promptly review
+10. UX designs have been reviewed and signed off by stakeholders with no major changes to be expected. These designs need to be in a state where the work can be scoped by the dashboard team, though not necessarily complete. Small details can be worked on as we start allowing the team to discover minor implementation details about the flow during development.
+11. API (REST and CRD) requirements for the feature should be clearly documented and reviewed with one of the advisors on the Dashboard team.
+12. Feature implementation document including API requirements and any high level implementation points that are critical should be created and reviewed by an advisor on the Dashboard team.
+13. Dependencies and backend component apis should be fully ready (including v2 Operator support built-in).  These features could be enabled via DevFlags if needed but should not be installed via scripts or yaml files provided by the backend team.
```
Comment (Griffin-Sullivan) [inline docs/definition-of-ready.md:18]:
```suggestion
13. Dependencies and backend component APIs should be fully ready (including v2 Operator support built-in).  These features could be enabled via DevFlags if needed but should not be installed via scripts or yaml files provided by the backend team.  
```

**Review by Griffin-Sullivan — COMMENTED** [review]
From the perspective of someone new who just went through all of the dev setup and contributing docs: LGTM

**Review by andrewballantyne — CHANGES_REQUESTED** [review]

**Comment by andrewballantyne** [comment]
@jessiehuff please update the text from @Griffin-Sullivan, we can look to merge this and iterate as needed.

**Review by andrewballantyne — APPROVED** [review]

**Pull request merged**